contracts/src/utils/Call.sol

```diff
@@ -22,4 +22,54 @@ library Call {
             }
         }
     }
+
+    /// @notice Use when you _really_ really _really_ don't trust the called
+    /// contract. This prevents the called contract from causing reversion of
+    /// the caller in as many ways as we can.
```

Comment:
We should understand how using zexcessivelySafeCall` can still fail? And potentially have tests for at least one case to make sure we don't poisen pill the inbound queue.

Reply:
Yeah, an unsafe call could be dangerous so we need to be very careful here. Except that I would also like to introduce white list control, something like the [SafeCallFilter](https://github.com/polkadot-fellows/runtimes/blob/2ef38e75b43b2b91d160936843fa8b3e4ffbf5fc/system-parachains/bridge-hubs/bridge-hub-kusama/src/xcm_config.rs#L135) but on Ethereum side.

Reply:
https://github.com/Snowfork/snowbridge/pull/1145/commits/ebc89866f4134dbd7c8e6aaf2f6ead33330a7fa5

Reply:
My thoughts about this:

1. I think `excessivelySafeCall` (used by Nomad) is too complicated for our needs. When issuing a call, we are only interested in whether the call succeeded or not. The `returndata` we can ignore. For this reason, I prefer that we use this safe call solution: https://github.com/ethereum-optimism/optimism/blob/b36eb5515cc2a34a15383b2eee488dbac83d6caf/packages/contracts-bedrock/src/libraries/SafeCall.sol#L12

2. I am optimistic we've accounted for all the ways a contract call can be unsafe. This includes guards against re-entrancy (nonce check), gas limits, and returnbomb attacks. Therefore I don't see a reason to include a safe call filter. If we think a call can still be unsafe, we need to understand why.

Reply:
> How does SafeCallFilter work? Will we have to restrict allowing transact to only explicitly specified contracts and/or functions?

It's not by us or require any sudo call from relaychain, it's a system control call from the Parachain sovereign through xcm to increase the legal transacts step by step totally in their own control.

In this way we can release the transact PR but disable it without introducing any risk.

Reply:
SafeCall ignore returndata https://github.com/Snowfork/snowbridge/pull/1145/commits/0a8dc1e425d495a1bfa217cea6dde520260519ec